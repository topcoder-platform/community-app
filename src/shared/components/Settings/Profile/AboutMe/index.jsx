/**
 * Child component of Settings/Profile/AboutMe renders "About Me" section of profile setting page.
 */
/* global document */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';

import { PrimaryButton, SecondaryButton } from 'topcoder-react-ui-kit';

import { getAllCountryObjects, getCountryObjFromAlpha3 } from 'utils/countries';

import Select from 'components/Select';
import DefaultPortrait from 'assets/images/ico-user-default.svg';

import Styles from './styles.scss';

const countries = getAllCountryObjects();

export default class AboutMe extends React.Component {
  constructor(props) {
    super(props);

    this.onChangeImage = this.onChangeImage.bind(this);
    this.onUploadPhoto = this.onUploadPhoto.bind(this);
    this.onDeletePhoto = this.onDeletePhoto.bind(this);

    this.onUpdateBio = this.onUpdateBio.bind(this);
    this.onUpdateCountry = this.onUpdateCountry.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.profileState.deletingPhoto && !nextProps.profileState.deletingPhoto) {
      document.querySelector('#change-image-input').value = null;
    }
  }

  onChangeImage(e) {
    e.preventDefault();
    if (this.props.profileState.uploadingPhoto) {
      return;
    }
    const fileInput = document.querySelector('#change-image-input');
    fileInput.click();
  }

  onUploadPhoto(e) {
    e.preventDefault();
    if (this.props.profileState.uploadingPhoto) {
      return;
    }
    const fileInput = document.querySelector('#change-image-input');
    const file = fileInput.files[0];
    this.props.uploadPhoto(this.props.handle, this.props.tokenV3, file);
  }

  onDeletePhoto(e) {
    e.preventDefault();
    if (this.props.profileState.deletingPhoto) {
      return;
    }
    const newProfile = _.clone(this.props.profile);
    delete newProfile.photoURL;
    delete newProfile.groups;
    newProfile.tracks = newProfile.tracks || [];
    this.props.deletePhoto(newProfile, this.props.tokenV3);
  }

  onUpdateBio(e) {
    e.preventDefault();
    const bio = document.querySelector('#profile-bio').value;
    this.props.onUpdateBio(bio);
  }

  onUpdateCountry(country) {
    if (country && country.alpha3) {
      this.props.onUpdateCountry(country.alpha3);
    }
  }

  render() {
    const {
      profile,
      profileState,
      bio,
      countryCode,
    } = this.props;

    const {
      uploadingPhoto,
      deletingPhoto,
    } = profileState;

    const userCountry = getCountryObjFromAlpha3(countryCode);

    return (
      <div className="settings-section">
        <div className="section-info">
          <h2>About Me</h2>
          <div className="description">The most important information that other community members should know about you.</div>
        </div>
        <div className="section-fields">
          <div styleName="image">
            <div className="form-label">your profile image</div>
            <div styleName="edit-image">
              {
                profile.photoURL &&
                <img alt="User" src={profile.photoURL} styleName="profile-circle" />
              }
              {
                !profile.photoURL &&
                <DefaultPortrait styleName="profile-circle" />
              }
              <div styleName="buttons">
                <PrimaryButton onClick={this.onChangeImage} disabled={uploadingPhoto || deletingPhoto} theme={{ button: Styles['file-upload'] }}>
                  {
                    uploadingPhoto && <i className="fa fa-spinner fa-spin" />
                  }
                  {
                    !uploadingPhoto && profile.photoURL && 'Change Image'
                  }
                  {
                    !uploadingPhoto && !profile.photoURL && 'Add Image'
                  }
                </PrimaryButton>
                <input type="file" name="image" onChange={this.onUploadPhoto} id="change-image-input" className="hidden" />
                {
                  profile.photoURL &&
                  <div>
                    <SecondaryButton
                      onClick={this.onDeletePhoto}
                      disabled={uploadingPhoto || deletingPhoto}
                      theme={{ button: Styles['file-delete'] }}
                    >
                      {
                        deletingPhoto && <i className="fa fa-spinner fa-spin" />
                      }
                      {
                        !deletingPhoto && 'Delete'
                      }
                    </SecondaryButton>
                  </div>
                }
              </div>
            </div>
          </div>
          <div styleName="country">
            <div className="form-label">Country to represent</div>
            <Select
              name="location"
              options={countries}
              value={userCountry}
              onChange={this.onUpdateCountry}
              placeholder="Country"
              matchPos="start"
              matchProp="name"
              labelKey="name"
              valueKey="name"
              clearable={false}
            />
          </div>
          <div styleName="bio">
            <div className="form-label">short bio
              <span className="char-count">{(bio && bio.length) || 0}
                <span className="grey">&nbsp;/ 256</span>
              </span>
            </div>
            <textarea
              name="bio"
              id="profile-bio"
              value={bio}
              onChange={this.onUpdateBio}
              maxLength="256"
              className="topcoder-input"
              placeholder="E.g., I'm a JS architect interested in creating new data interchange formats. I love sci-fi and riding my motorcycle."
            />
          </div>
        </div>
      </div>
    );
  }
}

AboutMe.defaultProps = {
  bio: '',
  countryCode: null,
};

AboutMe.propTypes = {
  handle: PT.string.isRequired,
  tokenV3: PT.string.isRequired,
  profile: PT.shape().isRequired,
  profileState: PT.shape().isRequired,
  bio: PT.string,
  countryCode: PT.string,
  onUpdateCountry: PT.func.isRequired,
  onUpdateBio: PT.func.isRequired,
  uploadPhoto: PT.func.isRequired,
  deletePhoto: PT.func.isRequired,
};

