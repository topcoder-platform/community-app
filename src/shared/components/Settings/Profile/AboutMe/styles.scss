@import "~styles/mixins";

.country {
  margin-top: 30px;

  input {
    width: 100%;
  }

  span.error-message {
    color: #ef3a3a;

    @include sofia-pro-regular;

    font-size: 10px;
    text-transform: uppercase;
  }
}

.image {
  .edit-image {
    display: flex;
    flex-direction: row;
    align-items: center;

    .profile-circle {
      margin-top: 10px;
      border-radius: 50%;
      border: 1px solid #d1d3d4;
      margin-right: 20px;
      width: 120px;
      height: 120px;
      padding: 2px;
    }

    .upload-image {
      display: flex;
      flex-direction: column;

      p {
        margin-top: 15px;
        margin-bottom: 10px;
      }

      input {
        padding: 0;
      }
    }

    .buttons {
      padding-top: 20px;

      .file-upload {
        position: relative;
        overflow: hidden;
        margin: 10px;
        margin-left: 0;
        margin-top: 20px;
        width: 120px;
        text-transform: uppercase;
        font-size: 12px;
        min-height: 30px;
        padding: 0;
      }

      .file-delete {
        width: 80px;
        margin-left: 0;
        text-transform: uppercase;
        font-size: 12px;
        min-height: 30px;
        padding: 0;

        &:hover {
          background: $tc-dark-blue;
        }
      }
    }
  }
}

.bio {
  margin-top: 30px;

  textarea {
    background: white;
    width: 100%;
    height: 115px;

    @include placeholder {
      color: $tc-gray-40;

      @include merriweather-sans-light;
    }
  }
}
