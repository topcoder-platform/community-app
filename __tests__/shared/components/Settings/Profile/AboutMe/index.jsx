import React from 'react';
import Renderer from 'react-test-renderer/shallow';

import AboutMe from 'components/Settings/Profile/AboutMe';

import userProfile from '../../__mocks__/user-profile.json';
import profileState from '../../__mocks__/profile-state.json';

const rnd = new Renderer();

it('renders about me section of profile setting page correctly', () => {
  rnd.render((<AboutMe
    handle=""
    tokenV3=""
    profile={userProfile}
    profileState={profileState}
    bio={userProfile.description}
    countryCode={userProfile.competitionCountryCode}
    onUpdateCountry={() => {}}
    onUpdateBio={() => {}}
    uploadPhoto={() => {}}
    deletePhoto={() => {}}
  />));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
