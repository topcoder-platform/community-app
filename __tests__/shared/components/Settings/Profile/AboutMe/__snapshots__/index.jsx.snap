// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders about me section of profile setting page correctly 1`] = `
<div
  className="settings-section"
>
  <div
    className="section-info"
  >
    <h2>
      About Me
    </h2>
    <div
      className="description"
    >
      The most important information that other community members should know about you.
    </div>
  </div>
  <div
    className="section-fields"
  >
    <div
      className="src-shared-components-Settings-Profile-AboutMe-___styles__image___2Pzzy"
    >
      <div
        className="form-label"
      >
        your profile image
      </div>
      <div
        className="src-shared-components-Settings-Profile-AboutMe-___styles__edit-image___3BNKa"
      >
        <img
          alt="User"
          className="src-shared-components-Settings-Profile-AboutMe-___styles__profile-circle___2aNdR"
          src="https://topcoder-prod-media.s3.amazonaws.com/member/profile/tcscoder-1525559661069.png"
        />
        <div
          className="src-shared-components-Settings-Profile-AboutMe-___styles__buttons___2PWXt"
        >
          <ThemedButton
            active={false}
            composeAdhocTheme="deeply"
            composeContextTheme="softly"
            disabled={false}
            enforceA={false}
            mapThemrProps={[Function]}
            onClick={[Function]}
            onMouseDown={null}
            openNewTab={false}
            replace={false}
            size={null}
            theme={
              Object {
                "button": "src-shared-components-Settings-Profile-AboutMe-styles___file-upload___1uQbWx",
              }
            }
            themePriority="adhoc-context-default"
            to={null}
            type="button"
          >
            Change Image
          </ThemedButton>
          <input
            className="hidden"
            id="change-image-input"
            name="image"
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </div>
    <div
      className="src-shared-components-Settings-Profile-AboutMe-___styles__country___cWUCk"
    >
      <div
        className="form-label"
      >
        Country to represent
      </div>
      <Select
        clearable={false}
        labelKey="name"
        matchPos="start"
        matchProp="name"
        name="location"
        onChange={[Function]}
        options={
          Array [
            Object {
              "alpha3": "AFG",
              "name": "Afghanistan",
            },
            Object {
              "alpha3": "ALB",
              "name": "Albania",
            },
            Object {
              "alpha3": "DZA",
              "name": "Algeria",
            },
            Object {
              "alpha3": "ASM",
              "name": "American Samoa",
            },
            Object {
              "alpha3": "AND",
              "name": "Andorra",
            },
            Object {
              "alpha3": "AGO",
              "name": "Angola",
            },
            Object {
              "alpha3": "AIA",
              "name": "Anguilla",
            },
            Object {
              "alpha3": "ATA",
              "name": "Antarctica",
            },
            Object {
              "alpha3": "ATG",
              "name": "Antigua and Barbuda",
            },
            Object {
              "alpha3": "ARG",
              "name": "Argentina",
            },
            Object {
              "alpha3": "ARM",
              "name": "Armenia",
            },
            Object {
              "alpha3": "ABW",
              "name": "Aruba",
            },
            Object {
              "alpha3": "AUS",
              "name": "Australia",
            },
            Object {
              "alpha3": "AUT",
              "name": "Austria",
            },
            Object {
              "alpha3": "AZE",
              "name": "Azerbaijan",
            },
            Object {
              "alpha3": "BHS",
              "name": "Bahamas",
            },
            Object {
              "alpha3": "BHR",
              "name": "Bahrain",
            },
            Object {
              "alpha3": "BGD",
              "name": "Bangladesh",
            },
            Object {
              "alpha3": "BRB",
              "name": "Barbados",
            },
            Object {
              "alpha3": "BLR",
              "name": "Belarus",
            },
            Object {
              "alpha3": "BEL",
              "name": "Belgium",
            },
            Object {
              "alpha3": "BLZ",
              "name": "Belize",
            },
            Object {
              "alpha3": "BEN",
              "name": "Benin",
            },
            Object {
              "alpha3": "BMU",
              "name": "Bermuda",
            },
            Object {
              "alpha3": "BTN",
              "name": "Bhutan",
            },
            Object {
              "alpha3": "BOL",
              "name": "Bolivia",
            },
            Object {
              "alpha3": "BIH",
              "name": "Bosnia and Herzegovina",
            },
            Object {
              "alpha3": "BWA",
              "name": "Botswana",
            },
            Object {
              "alpha3": "BVT",
              "name": "Bouvet Island",
            },
            Object {
              "alpha3": "BRA",
              "name": "Brazil",
            },
            Object {
              "alpha3": "IOT",
              "name": "British Indian Ocean Territory",
            },
            Object {
              "alpha3": "BRN",
              "name": "Brunei Darussalam",
            },
            Object {
              "alpha3": "BGR",
              "name": "Bulgaria",
            },
            Object {
              "alpha3": "BFA",
              "name": "Burkina Faso",
            },
            Object {
              "alpha3": "BDI",
              "name": "Burundi",
            },
            Object {
              "alpha3": "KHM",
              "name": "Cambodia",
            },
            Object {
              "alpha3": "CMR",
              "name": "Cameroon",
            },
            Object {
              "alpha3": "CAN",
              "name": "Canada",
            },
            Object {
              "alpha3": "CPV",
              "name": "Cape Verde",
            },
            Object {
              "alpha3": "CYM",
              "name": "Cayman Islands",
            },
            Object {
              "alpha3": "CAF",
              "name": "Central African Republic",
            },
            Object {
              "alpha3": "TCD",
              "name": "Chad",
            },
            Object {
              "alpha3": "CHL",
              "name": "Chile",
            },
            Object {
              "alpha3": "CHN",
              "name": "China",
            },
            Object {
              "alpha3": "CXR",
              "name": "Christmas Island",
            },
            Object {
              "alpha3": "CCK",
              "name": "Cocos (Keeling) Islands",
            },
            Object {
              "alpha3": "COL",
              "name": "Colombia",
            },
            Object {
              "alpha3": "COM",
              "name": "Comoros",
            },
            Object {
              "alpha3": "COG",
              "name": "Congo",
            },
            Object {
              "alpha3": "COD",
              "name": "Congo, the Democratic Republic of the",
            },
            Object {
              "alpha3": "COK",
              "name": "Cook Islands",
            },
            Object {
              "alpha3": "CRI",
              "name": "Costa Rica",
            },
            Object {
              "alpha3": "CIV",
              "name": "Cote D'Ivoire",
            },
            Object {
              "alpha3": "HRV",
              "name": "Croatia",
            },
            Object {
              "alpha3": "CUB",
              "name": "Cuba",
            },
            Object {
              "alpha3": "CYP",
              "name": "Cyprus",
            },
            Object {
              "alpha3": "CZE",
              "name": "Czech Republic",
            },
            Object {
              "alpha3": "DNK",
              "name": "Denmark",
            },
            Object {
              "alpha3": "DJI",
              "name": "Djibouti",
            },
            Object {
              "alpha3": "DMA",
              "name": "Dominica",
            },
            Object {
              "alpha3": "DOM",
              "name": "Dominican Republic",
            },
            Object {
              "alpha3": "ECU",
              "name": "Ecuador",
            },
            Object {
              "alpha3": "EGY",
              "name": "Egypt",
            },
            Object {
              "alpha3": "SLV",
              "name": "El Salvador",
            },
            Object {
              "alpha3": "GNQ",
              "name": "Equatorial Guinea",
            },
            Object {
              "alpha3": "ERI",
              "name": "Eritrea",
            },
            Object {
              "alpha3": "EST",
              "name": "Estonia",
            },
            Object {
              "alpha3": "ETH",
              "name": "Ethiopia",
            },
            Object {
              "alpha3": "FLK",
              "name": "Falkland Islands (Malvinas)",
            },
            Object {
              "alpha3": "FRO",
              "name": "Faroe Islands",
            },
            Object {
              "alpha3": "FJI",
              "name": "Fiji",
            },
            Object {
              "alpha3": "FIN",
              "name": "Finland",
            },
            Object {
              "alpha3": "FRA",
              "name": "France",
            },
            Object {
              "alpha3": "GUF",
              "name": "French Guiana",
            },
            Object {
              "alpha3": "PYF",
              "name": "French Polynesia",
            },
            Object {
              "alpha3": "ATF",
              "name": "French Southern Territories",
            },
            Object {
              "alpha3": "GAB",
              "name": "Gabon",
            },
            Object {
              "alpha3": "GMB",
              "name": "Gambia",
            },
            Object {
              "alpha3": "GEO",
              "name": "Georgia",
            },
            Object {
              "alpha3": "DEU",
              "name": "Germany",
            },
            Object {
              "alpha3": "GHA",
              "name": "Ghana",
            },
            Object {
              "alpha3": "GIB",
              "name": "Gibraltar",
            },
            Object {
              "alpha3": "GRC",
              "name": "Greece",
            },
            Object {
              "alpha3": "GRL",
              "name": "Greenland",
            },
            Object {
              "alpha3": "GRD",
              "name": "Grenada",
            },
            Object {
              "alpha3": "GLP",
              "name": "Guadeloupe",
            },
            Object {
              "alpha3": "GUM",
              "name": "Guam",
            },
            Object {
              "alpha3": "GTM",
              "name": "Guatemala",
            },
            Object {
              "alpha3": "GIN",
              "name": "Guinea",
            },
            Object {
              "alpha3": "GNB",
              "name": "Guinea-Bissau",
            },
            Object {
              "alpha3": "GUY",
              "name": "Guyana",
            },
            Object {
              "alpha3": "HTI",
              "name": "Haiti",
            },
            Object {
              "alpha3": "HMD",
              "name": "Heard Island and Mcdonald Islands",
            },
            Object {
              "alpha3": "VAT",
              "name": "Holy See (Vatican City State)",
            },
            Object {
              "alpha3": "HND",
              "name": "Honduras",
            },
            Object {
              "alpha3": "HKG",
              "name": "Hong Kong",
            },
            Object {
              "alpha3": "HUN",
              "name": "Hungary",
            },
            Object {
              "alpha3": "ISL",
              "name": "Iceland",
            },
            Object {
              "alpha3": "IND",
              "name": "India",
            },
            Object {
              "alpha3": "IDN",
              "name": "Indonesia",
            },
            Object {
              "alpha3": "IRN",
              "name": "Iran, Islamic Republic of",
            },
            Object {
              "alpha3": "IRQ",
              "name": "Iraq",
            },
            Object {
              "alpha3": "IRL",
              "name": "Ireland",
            },
            Object {
              "alpha3": "ISR",
              "name": "Israel",
            },
            Object {
              "alpha3": "ITA",
              "name": "Italy",
            },
            Object {
              "alpha3": "JAM",
              "name": "Jamaica",
            },
            Object {
              "alpha3": "JPN",
              "name": "Japan",
            },
            Object {
              "alpha3": "JOR",
              "name": "Jordan",
            },
            Object {
              "alpha3": "KAZ",
              "name": "Kazakhstan",
            },
            Object {
              "alpha3": "KEN",
              "name": "Kenya",
            },
            Object {
              "alpha3": "KIR",
              "name": "Kiribati",
            },
            Object {
              "alpha3": "PRK",
              "name": "North Korea",
            },
            Object {
              "alpha3": "KOR",
              "name": "South Korea",
            },
            Object {
              "alpha3": "KWT",
              "name": "Kuwait",
            },
            Object {
              "alpha3": "KGZ",
              "name": "Kyrgyzstan",
            },
            Object {
              "alpha3": "LAO",
              "name": "Lao People's Democratic Republic",
            },
            Object {
              "alpha3": "LVA",
              "name": "Latvia",
            },
            Object {
              "alpha3": "LBN",
              "name": "Lebanon",
            },
            Object {
              "alpha3": "LSO",
              "name": "Lesotho",
            },
            Object {
              "alpha3": "LBR",
              "name": "Liberia",
            },
            Object {
              "alpha3": "LBY",
              "name": "Libya",
            },
            Object {
              "alpha3": "LIE",
              "name": "Liechtenstein",
            },
            Object {
              "alpha3": "LTU",
              "name": "Lithuania",
            },
            Object {
              "alpha3": "LUX",
              "name": "Luxembourg",
            },
            Object {
              "alpha3": "MAC",
              "name": "Macao",
            },
            Object {
              "alpha3": "MKD",
              "name": "Macedonia, the Former Yugoslav Republic of",
            },
            Object {
              "alpha3": "MDG",
              "name": "Madagascar",
            },
            Object {
              "alpha3": "MWI",
              "name": "Malawi",
            },
            Object {
              "alpha3": "MYS",
              "name": "Malaysia",
            },
            Object {
              "alpha3": "MDV",
              "name": "Maldives",
            },
            Object {
              "alpha3": "MLI",
              "name": "Mali",
            },
            Object {
              "alpha3": "MLT",
              "name": "Malta",
            },
            Object {
              "alpha3": "MHL",
              "name": "Marshall Islands",
            },
            Object {
              "alpha3": "MTQ",
              "name": "Martinique",
            },
            Object {
              "alpha3": "MRT",
              "name": "Mauritania",
            },
            Object {
              "alpha3": "MUS",
              "name": "Mauritius",
            },
            Object {
              "alpha3": "MYT",
              "name": "Mayotte",
            },
            Object {
              "alpha3": "MEX",
              "name": "Mexico",
            },
            Object {
              "alpha3": "FSM",
              "name": "Micronesia, Federated States of",
            },
            Object {
              "alpha3": "MDA",
              "name": "Moldova, Republic of",
            },
            Object {
              "alpha3": "MCO",
              "name": "Monaco",
            },
            Object {
              "alpha3": "MNG",
              "name": "Mongolia",
            },
            Object {
              "alpha3": "MSR",
              "name": "Montserrat",
            },
            Object {
              "alpha3": "MAR",
              "name": "Morocco",
            },
            Object {
              "alpha3": "MOZ",
              "name": "Mozambique",
            },
            Object {
              "alpha3": "MMR",
              "name": "Myanmar",
            },
            Object {
              "alpha3": "NAM",
              "name": "Namibia",
            },
            Object {
              "alpha3": "NRU",
              "name": "Nauru",
            },
            Object {
              "alpha3": "NPL",
              "name": "Nepal",
            },
            Object {
              "alpha3": "NLD",
              "name": "Netherlands",
            },
            Object {
              "alpha3": "NCL",
              "name": "New Caledonia",
            },
            Object {
              "alpha3": "NZL",
              "name": "New Zealand",
            },
            Object {
              "alpha3": "NIC",
              "name": "Nicaragua",
            },
            Object {
              "alpha3": "NER",
              "name": "Niger",
            },
            Object {
              "alpha3": "NGA",
              "name": "Nigeria",
            },
            Object {
              "alpha3": "NIU",
              "name": "Niue",
            },
            Object {
              "alpha3": "NFK",
              "name": "Norfolk Island",
            },
            Object {
              "alpha3": "MNP",
              "name": "Northern Mariana Islands",
            },
            Object {
              "alpha3": "NOR",
              "name": "Norway",
            },
            Object {
              "alpha3": "OMN",
              "name": "Oman",
            },
            Object {
              "alpha3": "PAK",
              "name": "Pakistan",
            },
            Object {
              "alpha3": "PLW",
              "name": "Palau",
            },
            Object {
              "alpha3": "PSE",
              "name": "Palestinian Territory, Occupied",
            },
            Object {
              "alpha3": "PAN",
              "name": "Panama",
            },
            Object {
              "alpha3": "PNG",
              "name": "Papua New Guinea",
            },
            Object {
              "alpha3": "PRY",
              "name": "Paraguay",
            },
            Object {
              "alpha3": "PER",
              "name": "Peru",
            },
            Object {
              "alpha3": "PHL",
              "name": "Philippines",
            },
            Object {
              "alpha3": "PCN",
              "name": "Pitcairn",
            },
            Object {
              "alpha3": "POL",
              "name": "Poland",
            },
            Object {
              "alpha3": "PRT",
              "name": "Portugal",
            },
            Object {
              "alpha3": "PRI",
              "name": "Puerto Rico",
            },
            Object {
              "alpha3": "QAT",
              "name": "Qatar",
            },
            Object {
              "alpha3": "REU",
              "name": "Reunion",
            },
            Object {
              "alpha3": "ROU",
              "name": "Romania",
            },
            Object {
              "alpha3": "RUS",
              "name": "Russian Federation",
            },
            Object {
              "alpha3": "RWA",
              "name": "Rwanda",
            },
            Object {
              "alpha3": "SHN",
              "name": "Saint Helena",
            },
            Object {
              "alpha3": "KNA",
              "name": "Saint Kitts and Nevis",
            },
            Object {
              "alpha3": "LCA",
              "name": "Saint Lucia",
            },
            Object {
              "alpha3": "SPM",
              "name": "Saint Pierre and Miquelon",
            },
            Object {
              "alpha3": "VCT",
              "name": "Saint Vincent and the Grenadines",
            },
            Object {
              "alpha3": "WSM",
              "name": "Samoa",
            },
            Object {
              "alpha3": "SMR",
              "name": "San Marino",
            },
            Object {
              "alpha3": "STP",
              "name": "Sao Tome and Principe",
            },
            Object {
              "alpha3": "SAU",
              "name": "Saudi Arabia",
            },
            Object {
              "alpha3": "SEN",
              "name": "Senegal",
            },
            Object {
              "alpha3": "SYC",
              "name": "Seychelles",
            },
            Object {
              "alpha3": "SLE",
              "name": "Sierra Leone",
            },
            Object {
              "alpha3": "SGP",
              "name": "Singapore",
            },
            Object {
              "alpha3": "SVK",
              "name": "Slovakia",
            },
            Object {
              "alpha3": "SVN",
              "name": "Slovenia",
            },
            Object {
              "alpha3": "SLB",
              "name": "Solomon Islands",
            },
            Object {
              "alpha3": "SOM",
              "name": "Somalia",
            },
            Object {
              "alpha3": "ZAF",
              "name": "South Africa",
            },
            Object {
              "alpha3": "SGS",
              "name": "South Georgia and the South Sandwich Islands",
            },
            Object {
              "alpha3": "ESP",
              "name": "Spain",
            },
            Object {
              "alpha3": "LKA",
              "name": "Sri Lanka",
            },
            Object {
              "alpha3": "SDN",
              "name": "Sudan",
            },
            Object {
              "alpha3": "SUR",
              "name": "Suriname",
            },
            Object {
              "alpha3": "SJM",
              "name": "Svalbard and Jan Mayen",
            },
            Object {
              "alpha3": "SWZ",
              "name": "Swaziland",
            },
            Object {
              "alpha3": "SWE",
              "name": "Sweden",
            },
            Object {
              "alpha3": "CHE",
              "name": "Switzerland",
            },
            Object {
              "alpha3": "SYR",
              "name": "Syrian Arab Republic",
            },
            Object {
              "alpha3": "TWN",
              "name": "Taiwan",
            },
            Object {
              "alpha3": "TJK",
              "name": "Tajikistan",
            },
            Object {
              "alpha3": "TZA",
              "name": "Tanzania, United Republic of",
            },
            Object {
              "alpha3": "THA",
              "name": "Thailand",
            },
            Object {
              "alpha3": "TLS",
              "name": "Timor-Leste",
            },
            Object {
              "alpha3": "TGO",
              "name": "Togo",
            },
            Object {
              "alpha3": "TKL",
              "name": "Tokelau",
            },
            Object {
              "alpha3": "TON",
              "name": "Tonga",
            },
            Object {
              "alpha3": "TTO",
              "name": "Trinidad and Tobago",
            },
            Object {
              "alpha3": "TUN",
              "name": "Tunisia",
            },
            Object {
              "alpha3": "TUR",
              "name": "Turkey",
            },
            Object {
              "alpha3": "TKM",
              "name": "Turkmenistan",
            },
            Object {
              "alpha3": "TCA",
              "name": "Turks and Caicos Islands",
            },
            Object {
              "alpha3": "TUV",
              "name": "Tuvalu",
            },
            Object {
              "alpha3": "UGA",
              "name": "Uganda",
            },
            Object {
              "alpha3": "UKR",
              "name": "Ukraine",
            },
            Object {
              "alpha3": "ARE",
              "name": "United Arab Emirates",
            },
            Object {
              "alpha3": "GBR",
              "name": "United Kingdom",
            },
            Object {
              "alpha3": "USA",
              "name": "United States of America",
            },
            Object {
              "alpha3": "UMI",
              "name": "United States Minor Outlying Islands",
            },
            Object {
              "alpha3": "URY",
              "name": "Uruguay",
            },
            Object {
              "alpha3": "UZB",
              "name": "Uzbekistan",
            },
            Object {
              "alpha3": "VUT",
              "name": "Vanuatu",
            },
            Object {
              "alpha3": "VEN",
              "name": "Venezuela",
            },
            Object {
              "alpha3": "VNM",
              "name": "Viet Nam",
            },
            Object {
              "alpha3": "VGB",
              "name": "Virgin Islands, British",
            },
            Object {
              "alpha3": "VIR",
              "name": "Virgin Islands, U.S.",
            },
            Object {
              "alpha3": "WLF",
              "name": "Wallis and Futuna",
            },
            Object {
              "alpha3": "ESH",
              "name": "Western Sahara",
            },
            Object {
              "alpha3": "YEM",
              "name": "Yemen",
            },
            Object {
              "alpha3": "ZMB",
              "name": "Zambia",
            },
            Object {
              "alpha3": "ZWE",
              "name": "Zimbabwe",
            },
            Object {
              "alpha3": "ALA",
              "name": "Åland Islands",
            },
            Object {
              "alpha3": "BES",
              "name": "Bonaire, Sint Eustatius and Saba",
            },
            Object {
              "alpha3": "CUW",
              "name": "Curaçao",
            },
            Object {
              "alpha3": "GGY",
              "name": "Guernsey",
            },
            Object {
              "alpha3": "IMN",
              "name": "Isle of Man",
            },
            Object {
              "alpha3": "JEY",
              "name": "Jersey",
            },
            Object {
              "alpha3": "MNE",
              "name": "Montenegro",
            },
            Object {
              "alpha3": "BLM",
              "name": "Saint Barthélemy",
            },
            Object {
              "alpha3": "MAF",
              "name": "Saint Martin (French part)",
            },
            Object {
              "alpha3": "SRB",
              "name": "Serbia",
            },
            Object {
              "alpha3": "SXM",
              "name": "Sint Maarten (Dutch part)",
            },
            Object {
              "alpha3": "SSD",
              "name": "South Sudan",
            },
            Object {
              "alpha3": "XKX",
              "name": "Kosovo",
            },
          ]
        }
        placeholder="Country"
        selectRef={[Function]}
        value={
          Object {
            "alpha3": "CHN",
            "name": "China",
          }
        }
        valueKey="name"
      />
    </div>
    <div
      className="src-shared-components-Settings-Profile-AboutMe-___styles__bio___2xDtf"
    >
      <div
        className="form-label"
      >
        short bio
        <span
          className="char-count"
        >
          5
          <span
            className="grey"
          >
             / 256
          </span>
        </span>
      </div>
      <textarea
        className="topcoder-input"
        id="profile-bio"
        maxLength="256"
        name="bio"
        onChange={[Function]}
        placeholder="E.g., I'm a JS architect interested in creating new data interchange formats. I love sci-fi and riding my motorcycle."
        value="Hello"
      />
    </div>
  </div>
</div>
`;
